### ISSUE TITLE
ParameterizedType is not supported, causing errors with Ash.

### ISSUE THREAD

**Issue by vkryukov** [elixir-sqlite/ecto_sqlite3#158]
Hello,

it looks like the `Codec.float_decode` [does not](https://github.com/elixir-sqlite/ecto_sqlite3/blob/9b03747c44a71a0aafb8e6821bb7d290fb08be97/lib/ecto/adapters/sqlite3/codec.ex#L21) support parameterized types, which causes crashes when [using SQLite as a backend for Ash](https://github.com/ash-project/ash_sqlite/issues/114), as Ash uses a parameterized type for floats (to enable min/max limits, for example).

This [thread on ElixirForum](https://elixirforum.com/t/argumenterror-cannot-load-0-9-as-type-ash-type-float-ectotype/68274/4) adds more context.

It would be great to add this support. At the moment I don't fully understand the boundary in responsibilities between `ash_sqlite3` and `ecto_sqlite3`, but Ash developers seem to think that the issue should be fixed upstream.

Thank you for enabling Sqlite support for Elixir, and please let me know if you need any additional information.

**Comment by vkryukov** [comment]
It's seems that replacing
```elixir
  def float_decode(_), do: :error
```
with 
```
def float_decode(x), do: {:ok, x}
```
[as they do](https://github.com/elixir-ecto/ecto_sql/blob/df6732a7f5fcd50aa2ddc6a09b2526038c763105/lib/ecto/adapters/myxql.ex#L169) in `myxql` fixes the issue for me. I'm unsure if that's the right solution, but I can send a PR if it is.

**Comment by warmwaffles** [comment]
Is the issue because the value _is_ a float? IE a guard for `is_float(x)` entry would work. I know in myxql just throws its hands up and passes it along un-coerced, but I personally believe that is incorrect behavior.

**Comment by warmwaffles** [comment]
Fixed and released under `v0.18.0`.

**Comment by vkryukov** [comment]
Wow, that was fast! Thank you so much @warmwaffles !
